// @sec-ant/zxing-wasm/dist/full/index.js
import { g as ci, r as vi, d as N, a as di, w as hi, c as J } from "../core-bcd69edc.js";
import { Z as Pi, b as Ti, p as Ai, s as Ei } from "../core-bcd69edc.js";
var De = (() => {
  var W = import.meta.url;
  return function(V = {}) {
    var f = V, B, K;
    f.ready = new Promise((e, r) => {
      B = e, K = r;
    });
    var ve = Object.assign({}, f), de = "./this.program", ze = typeof window == "object", he = typeof importScripts == "function";
    typeof process == "object" && typeof process.versions == "object" && process.versions.node;
    var M = "";
    function Ur(e) {
      return f.locateFile ? f.locateFile(e, M) : M + e;
    }
    var ke;
    (ze || he) && (he ? M = self.location.href : typeof document < "u" && document.currentScript && (M = document.currentScript.src), W && (M = W), M.indexOf("blob:") !== 0 ? M = M.substr(0, M.replace(/[?#].*/, "").lastIndexOf("/") + 1) : M = "", he && (ke = (e) => {
      var r = new XMLHttpRequest();
      return r.open("GET", e, !1), r.responseType = "arraybuffer", r.send(null), new Uint8Array(r.response);
    })), f.print || console.log.bind(console);
    var re = f.printErr || console.error.bind(console);
    Object.assign(f, ve), ve = null, f.arguments && f.arguments, f.thisProgram && (de = f.thisProgram), f.quit && f.quit;
    var te;
    f.wasmBinary && (te = f.wasmBinary), typeof WebAssembly != "object" && ae("no native wasm support detected");
    var pe, Xe = !1;
    function xr(e, r) {
      e || ae(r);
    }
    var H, k, ne, _e, R, P, Ze, Ge;
    function qe() {
      var e = pe.buffer;
      f.HEAP8 = H = new Int8Array(e), f.HEAP16 = ne = new Int16Array(e), f.HEAPU8 = k = new Uint8Array(e), f.HEAPU16 = _e = new Uint16Array(e), f.HEAP32 = R = new Int32Array(e), f.HEAPU32 = P = new Uint32Array(e), f.HEAPF32 = Ze = new Float32Array(e), f.HEAPF64 = Ge = new Float64Array(e);
    }
    var Je = [], Ke = [], Qe = [];
    function Yr() {
      if (f.preRun)
        for (typeof f.preRun == "function" && (f.preRun = [f.preRun]); f.preRun.length; )
          Lr(f.preRun.shift());
      Oe(Je);
    }
    function Vr() {
      Oe(Ke);
    }
    function Br() {
      if (f.postRun)
        for (typeof f.postRun == "function" && (f.postRun = [f.postRun]); f.postRun.length; )
          zr(f.postRun.shift());
      Oe(Qe);
    }
    function Lr(e) {
      Je.unshift(e);
    }
    function Nr(e) {
      Ke.unshift(e);
    }
    function zr(e) {
      Qe.unshift(e);
    }
    var z = 0, ie = null;
    function Xr(e) {
      z++, f.monitorRunDependencies && f.monitorRunDependencies(z);
    }
    function Zr(e) {
      if (z--, f.monitorRunDependencies && f.monitorRunDependencies(z), z == 0 && ie) {
        var r = ie;
        ie = null, r();
      }
    }
    function ae(e) {
      f.onAbort && f.onAbort(e), e = "Aborted(" + e + ")", re(e), Xe = !0, e += ". Build with -sASSERTIONS for more info.";
      var r = new WebAssembly.RuntimeError(e);
      throw K(r), r;
    }
    var Gr = "data:application/octet-stream;base64,", er = (e) => e.startsWith(Gr), X;
    f.locateFile ? (X = "zxing_full.wasm", er(X) || (X = Ur(X))) : X = new URL("/full/zxing_full.wasm", self.location).href;
    function rr(e) {
      if (e == X && te)
        return new Uint8Array(te);
      if (ke)
        return ke(e);
      throw "both async and sync fetching of the wasm failed";
    }
    function qr(e) {
      return !te && (ze || he) && typeof fetch == "function" ? fetch(e, { credentials: "same-origin" }).then((r) => {
        if (!r.ok)
          throw "failed to load wasm binary file at '" + e + "'";
        return r.arrayBuffer();
      }).catch(() => rr(e)) : Promise.resolve().then(() => rr(e));
    }
    function tr(e, r, t) {
      return qr(e).then((n) => WebAssembly.instantiate(n, r)).then((n) => n).then(t, (n) => {
        re(`failed to asynchronously prepare wasm: ${n}`), ae(n);
      });
    }
    function Jr(e, r, t, n) {
      return !e && typeof WebAssembly.instantiateStreaming == "function" && !er(r) && typeof fetch == "function" ? fetch(r, { credentials: "same-origin" }).then((i) => {
        var a = WebAssembly.instantiateStreaming(i, t);
        return a.then(n, function(s) {
          return re(`wasm streaming compile failed: ${s}`), re("falling back to ArrayBuffer instantiation"), tr(r, t, n);
        });
      }) : tr(r, t, n);
    }
    function Kr() {
      var e = { a: Dn };
      function r(n, i) {
        return T = n.exports, pe = T.ra, qe(), _r = T.va, Nr(T.sa), Zr(), T;
      }
      Xr();
      function t(n) {
        r(n.instance);
      }
      if (f.instantiateWasm)
        try {
          return f.instantiateWasm(e, r);
        } catch (n) {
          re(`Module.instantiateWasm callback failed with error: ${n}`), K(n);
        }
      return Jr(te, X, e, t).catch(K), {};
    }
    var Oe = (e) => {
      for (; e.length > 0; )
        e.shift()(f);
    };
    f.noExitRuntime;
    var ge = [], ye = 0, Qr = (e) => {
      var r = new me(e);
      return r.get_caught() || (r.set_caught(!0), ye--), r.set_rethrown(!1), ge.push(r), Or(r.excPtr), r.get_exception_ptr();
    }, I = 0, et = () => {
      _(0, 0);
      var e = ge.pop();
      kr(e.excPtr), I = 0;
    };
    function me(e) {
      this.excPtr = e, this.ptr = e - 24, this.set_type = function(r) {
        P[this.ptr + 4 >> 2] = r;
      }, this.get_type = function() {
        return P[this.ptr + 4 >> 2];
      }, this.set_destructor = function(r) {
        P[this.ptr + 8 >> 2] = r;
      }, this.get_destructor = function() {
        return P[this.ptr + 8 >> 2];
      }, this.set_caught = function(r) {
        r = r ? 1 : 0, H[this.ptr + 12 >> 0] = r;
      }, this.get_caught = function() {
        return H[this.ptr + 12 >> 0] != 0;
      }, this.set_rethrown = function(r) {
        r = r ? 1 : 0, H[this.ptr + 13 >> 0] = r;
      }, this.get_rethrown = function() {
        return H[this.ptr + 13 >> 0] != 0;
      }, this.init = function(r, t) {
        this.set_adjusted_ptr(0), this.set_type(r), this.set_destructor(t);
      }, this.set_adjusted_ptr = function(r) {
        P[this.ptr + 16 >> 2] = r;
      }, this.get_adjusted_ptr = function() {
        return P[this.ptr + 16 >> 2];
      }, this.get_exception_ptr = function() {
        var r = Wr(this.get_type());
        if (r)
          return P[this.excPtr >> 2];
        var t = this.get_adjusted_ptr();
        return t !== 0 ? t : this.excPtr;
      };
    }
    var rt = (e) => {
      throw I || (I = e), I;
    }, je = (e) => {
      var r = I;
      if (!r)
        return ce(0), 0;
      var t = new me(r);
      t.set_adjusted_ptr(r);
      var n = t.get_type();
      if (!n)
        return ce(0), r;
      for (var i in e) {
        var a = e[i];
        if (a === 0 || a === n)
          break;
        var s = t.ptr + 16;
        if (jr(a, n, s))
          return ce(a), r;
      }
      return ce(n), r;
    }, tt = () => je([]), nt = (e) => je([e]), it = (e, r) => je([e, r]), at = (e) => {
      var r = new me(e).get_exception_ptr();
      return r;
    }, st = () => {
      var e = ge.pop();
      e || ae("no exception to throw");
      var r = e.excPtr;
      throw e.get_rethrown() || (ge.push(e), e.set_rethrown(!0), e.set_caught(!1), ye++), I = r, I;
    }, ot = (e, r, t) => {
      var n = new me(e);
      throw n.init(r, t), I = e, ye++, I;
    }, ut = () => ye, $e = {}, nr = (e) => {
      for (; e.length; ) {
        var r = e.pop(), t = e.pop();
        t(r);
      }
    };
    function We(e) {
      return this.fromWireType(R[e >> 2]);
    }
    var Q = {}, Z = {}, be = {}, ir, we = (e) => {
      throw new ir(e);
    }, G = (e, r, t) => {
      e.forEach(function(o) {
        be[o] = r;
      });
      function n(o) {
        var u = t(o);
        u.length !== e.length && we("Mismatched type converter count");
        for (var l = 0; l < e.length; ++l)
          U(e[l], u[l]);
      }
      var i = new Array(r.length), a = [], s = 0;
      r.forEach((o, u) => {
        Z.hasOwnProperty(o) ? i[u] = Z[o] : (a.push(o), Q.hasOwnProperty(o) || (Q[o] = []), Q[o].push(() => {
          i[u] = Z[o], ++s, s === a.length && n(i);
        }));
      }), a.length === 0 && n(i);
    }, ft = (e) => {
      var r = $e[e];
      delete $e[e];
      var t = r.rawConstructor, n = r.rawDestructor, i = r.fields, a = i.map((s) => s.getterReturnType).concat(i.map((s) => s.setterArgumentType));
      G([e], a, (s) => {
        var o = {};
        return i.forEach((u, l) => {
          var v = u.fieldName, h = s[l], p = u.getter, $ = u.getterContext, w = s[l + i.length], F = u.setter, C = u.setterContext;
          o[v] = { read: (S) => h.fromWireType(p($, S)), write: (S, c) => {
            var d = [];
            F(C, S, w.toWireType(d, c)), nr(d);
          } };
        }), [{ name: r.name, fromWireType: (u) => {
          var l = {};
          for (var v in o)
            l[v] = o[v].read(u);
          return n(u), l;
        }, toWireType: (u, l) => {
          for (var v in o)
            if (!(v in l))
              throw new TypeError(`Missing field: "${v}"`);
          var h = t();
          for (v in o)
            o[v].write(h, l[v]);
          return u !== null && u.push(n, h), h;
        }, argPackAdvance: x, readValueFromPointer: We, destructorFunction: n }];
      });
    }, lt = (e, r, t, n, i) => {
    }, ct = () => {
      for (var e = new Array(256), r = 0; r < 256; ++r)
        e[r] = String.fromCharCode(r);
      ar = e;
    }, ar, D = (e) => {
      for (var r = "", t = e; k[t]; )
        r += ar[k[t++]];
      return r;
    }, ee, b = (e) => {
      throw new ee(e);
    };
    function vt(e, r, t = {}) {
      var n = r.name;
      if (e || b(`type "${n}" must have a positive integer typeid pointer`), Z.hasOwnProperty(e)) {
        if (t.ignoreDuplicateRegistrations)
          return;
        b(`Cannot register type '${n}' twice`);
      }
      if (Z[e] = r, delete be[e], Q.hasOwnProperty(e)) {
        var i = Q[e];
        delete Q[e], i.forEach((a) => a());
      }
    }
    function U(e, r, t = {}) {
      if (!("argPackAdvance" in r))
        throw new TypeError("registerType registeredInstance requires argPackAdvance");
      return vt(e, r, t);
    }
    var x = 8, dt = (e, r, t, n) => {
      r = D(r), U(e, { name: r, fromWireType: function(i) {
        return !!i;
      }, toWireType: function(i, a) {
        return a ? t : n;
      }, argPackAdvance: x, readValueFromPointer: function(i) {
        return this.fromWireType(k[i]);
      }, destructorFunction: null });
    }, ht = (e) => ({ count: e.count, deleteScheduled: e.deleteScheduled, preservePointerOnDelete: e.preservePointerOnDelete, ptr: e.ptr, ptrType: e.ptrType, smartPtr: e.smartPtr, smartPtrType: e.smartPtrType }), Me = (e) => {
      function r(t) {
        return t.$$.ptrType.registeredClass.name;
      }
      b(r(e) + " instance already deleted");
    }, He = !1, sr = (e) => {
    }, pt = (e) => {
      e.smartPtr ? e.smartPtrType.rawDestructor(e.smartPtr) : e.ptrType.registeredClass.rawDestructor(e.ptr);
    }, or = (e) => {
      e.count.value -= 1;
      var r = e.count.value === 0;
      r && pt(e);
    }, ur = (e, r, t) => {
      if (r === t)
        return e;
      if (t.baseClass === void 0)
        return null;
      var n = ur(e, r, t.baseClass);
      return n === null ? null : t.downcast(n);
    }, fr = {}, _t = () => Object.keys(ue).length, gt = () => {
      var e = [];
      for (var r in ue)
        ue.hasOwnProperty(r) && e.push(ue[r]);
      return e;
    }, se = [], Ie = () => {
      for (; se.length; ) {
        var e = se.pop();
        e.$$.deleteScheduled = !1, e.delete();
      }
    }, oe, yt = (e) => {
      oe = e, se.length && oe && oe(Ie);
    }, mt = () => {
      f.getInheritedInstanceCount = _t, f.getLiveInheritedInstances = gt, f.flushPendingDeletes = Ie, f.setDelayFunction = yt;
    }, ue = {}, $t = (e, r) => {
      for (r === void 0 && b("ptr should not be undefined"); e.baseClass; )
        r = e.upcast(r), e = e.baseClass;
      return r;
    }, bt = (e, r) => (r = $t(e, r), ue[r]), Ce = (e, r) => {
      (!r.ptrType || !r.ptr) && we("makeClassHandle requires ptr and ptrType");
      var t = !!r.smartPtrType, n = !!r.smartPtr;
      return t !== n && we("Both smartPtrType and smartPtr must be specified"), r.count = { value: 1 }, fe(Object.create(e, { $$: { value: r } }));
    };
    function wt(e) {
      var r = this.getPointee(e);
      if (!r)
        return this.destructor(e), null;
      var t = bt(this.registeredClass, r);
      if (t !== void 0) {
        if (t.$$.count.value === 0)
          return t.$$.ptr = r, t.$$.smartPtr = e, t.clone();
        var n = t.clone();
        return this.destructor(e), n;
      }
      function i() {
        return this.isSmartPointer ? Ce(this.registeredClass.instancePrototype, { ptrType: this.pointeeType, ptr: r, smartPtrType: this, smartPtr: e }) : Ce(this.registeredClass.instancePrototype, { ptrType: this, ptr: e });
      }
      var a = this.registeredClass.getActualType(r), s = fr[a];
      if (!s)
        return i.call(this);
      var o;
      this.isConst ? o = s.constPointerType : o = s.pointerType;
      var u = ur(r, this.registeredClass, o.registeredClass);
      return u === null ? i.call(this) : this.isSmartPointer ? Ce(o.registeredClass.instancePrototype, { ptrType: o, ptr: u, smartPtrType: this, smartPtr: e }) : Ce(o.registeredClass.instancePrototype, { ptrType: o, ptr: u });
    }
    var fe = (e) => typeof FinalizationRegistry > "u" ? (fe = (r) => r, e) : (He = new FinalizationRegistry((r) => {
      or(r.$$);
    }), fe = (r) => {
      var t = r.$$, n = !!t.smartPtr;
      if (n) {
        var i = { $$: t };
        He.register(r, i, r);
      }
      return r;
    }, sr = (r) => He.unregister(r), fe(e)), Ct = () => {
      Object.assign(Pe.prototype, { isAliasOf(e) {
        if (!(this instanceof Pe) || !(e instanceof Pe))
          return !1;
        var r = this.$$.ptrType.registeredClass, t = this.$$.ptr;
        e.$$ = e.$$;
        for (var n = e.$$.ptrType.registeredClass, i = e.$$.ptr; r.baseClass; )
          t = r.upcast(t), r = r.baseClass;
        for (; n.baseClass; )
          i = n.upcast(i), n = n.baseClass;
        return r === n && t === i;
      }, clone() {
        if (this.$$.ptr || Me(this), this.$$.preservePointerOnDelete)
          return this.$$.count.value += 1, this;
        var e = fe(Object.create(Object.getPrototypeOf(this), { $$: { value: ht(this.$$) } }));
        return e.$$.count.value += 1, e.$$.deleteScheduled = !1, e;
      }, delete() {
        this.$$.ptr || Me(this), this.$$.deleteScheduled && !this.$$.preservePointerOnDelete && b("Object already scheduled for deletion"), sr(this), or(this.$$), this.$$.preservePointerOnDelete || (this.$$.smartPtr = void 0, this.$$.ptr = void 0);
      }, isDeleted() {
        return !this.$$.ptr;
      }, deleteLater() {
        return this.$$.ptr || Me(this), this.$$.deleteScheduled && !this.$$.preservePointerOnDelete && b("Object already scheduled for deletion"), se.push(this), se.length === 1 && oe && oe(Ie), this.$$.deleteScheduled = !0, this;
      } });
    };
    function Pe() {
    }
    var Pt = 48, Tt = 57, lr = (e) => {
      if (e === void 0)
        return "_unknown";
      e = e.replace(/[^a-zA-Z0-9_]/g, "$");
      var r = e.charCodeAt(0);
      return r >= Pt && r <= Tt ? `_${e}` : e;
    };
    function cr(e, r) {
      return e = lr(e), { [e]: function() {
        return r.apply(this, arguments);
      } }[e];
    }
    var vr = (e, r, t) => {
      if (e[r].overloadTable === void 0) {
        var n = e[r];
        e[r] = function() {
          return e[r].overloadTable.hasOwnProperty(arguments.length) || b(`Function '${t}' called with an invalid number of arguments (${arguments.length}) - expects one of (${e[r].overloadTable})!`), e[r].overloadTable[arguments.length].apply(this, arguments);
        }, e[r].overloadTable = [], e[r].overloadTable[n.argCount] = n;
      }
    }, dr = (e, r, t) => {
      f.hasOwnProperty(e) ? ((t === void 0 || f[e].overloadTable !== void 0 && f[e].overloadTable[t] !== void 0) && b(`Cannot register public name '${e}' twice`), vr(f, e, e), f.hasOwnProperty(t) && b(`Cannot register multiple overloads of a function with the same number of arguments (${t})!`), f[e].overloadTable[t] = r) : (f[e] = r, t !== void 0 && (f[e].numArguments = t));
    };
    function At(e, r, t, n, i, a, s, o) {
      this.name = e, this.constructor = r, this.instancePrototype = t, this.rawDestructor = n, this.baseClass = i, this.getActualType = a, this.upcast = s, this.downcast = o, this.pureVirtualFunctions = [];
    }
    var Ue = (e, r, t) => {
      for (; r !== t; )
        r.upcast || b(`Expected null or instance of ${t.name}, got an instance of ${r.name}`), e = r.upcast(e), r = r.baseClass;
      return e;
    };
    function Et(e, r) {
      if (r === null)
        return this.isReference && b(`null is not a valid ${this.name}`), 0;
      r.$$ || b(`Cannot pass "${Ve(r)}" as a ${this.name}`), r.$$.ptr || b(`Cannot pass deleted object as a pointer of type ${this.name}`);
      var t = r.$$.ptrType.registeredClass, n = Ue(r.$$.ptr, t, this.registeredClass);
      return n;
    }
    function Ft(e, r) {
      var t;
      if (r === null)
        return this.isReference && b(`null is not a valid ${this.name}`), this.isSmartPointer ? (t = this.rawConstructor(), e !== null && e.push(this.rawDestructor, t), t) : 0;
      r.$$ || b(`Cannot pass "${Ve(r)}" as a ${this.name}`), r.$$.ptr || b(`Cannot pass deleted object as a pointer of type ${this.name}`), !this.isConst && r.$$.ptrType.isConst && b(`Cannot convert argument of type ${r.$$.smartPtrType ? r.$$.smartPtrType.name : r.$$.ptrType.name} to parameter type ${this.name}`);
      var n = r.$$.ptrType.registeredClass;
      if (t = Ue(r.$$.ptr, n, this.registeredClass), this.isSmartPointer)
        switch (r.$$.smartPtr === void 0 && b("Passing raw pointer to smart pointer is illegal"), this.sharingPolicy) {
          case 0:
            r.$$.smartPtrType === this ? t = r.$$.smartPtr : b(`Cannot convert argument of type ${r.$$.smartPtrType ? r.$$.smartPtrType.name : r.$$.ptrType.name} to parameter type ${this.name}`);
            break;
          case 1:
            t = r.$$.smartPtr;
            break;
          case 2:
            if (r.$$.smartPtrType === this)
              t = r.$$.smartPtr;
            else {
              var i = r.clone();
              t = this.rawShare(t, L.toHandle(() => i.delete())), e !== null && e.push(this.rawDestructor, t);
            }
            break;
          default:
            b("Unsupporting sharing policy");
        }
      return t;
    }
    function St(e, r) {
      if (r === null)
        return this.isReference && b(`null is not a valid ${this.name}`), 0;
      r.$$ || b(`Cannot pass "${Ve(r)}" as a ${this.name}`), r.$$.ptr || b(`Cannot pass deleted object as a pointer of type ${this.name}`), r.$$.ptrType.isConst && b(`Cannot convert argument of type ${r.$$.ptrType.name} to parameter type ${this.name}`);
      var t = r.$$.ptrType.registeredClass, n = Ue(r.$$.ptr, t, this.registeredClass);
      return n;
    }
    function hr(e) {
      return this.fromWireType(P[e >> 2]);
    }
    var Rt = () => {
      Object.assign(Te.prototype, { getPointee(e) {
        return this.rawGetPointee && (e = this.rawGetPointee(e)), e;
      }, destructor(e) {
        this.rawDestructor && this.rawDestructor(e);
      }, argPackAdvance: x, readValueFromPointer: hr, deleteObject(e) {
        e !== null && e.delete();
      }, fromWireType: wt });
    };
    function Te(e, r, t, n, i, a, s, o, u, l, v) {
      this.name = e, this.registeredClass = r, this.isReference = t, this.isConst = n, this.isSmartPointer = i, this.pointeeType = a, this.sharingPolicy = s, this.rawGetPointee = o, this.rawConstructor = u, this.rawShare = l, this.rawDestructor = v, !i && r.baseClass === void 0 ? n ? (this.toWireType = Et, this.destructorFunction = null) : (this.toWireType = St, this.destructorFunction = null) : this.toWireType = Ft;
    }
    var pr = (e, r, t) => {
      f.hasOwnProperty(e) || we("Replacing nonexistant public symbol"), f[e].overloadTable !== void 0 && t !== void 0 ? f[e].overloadTable[t] = r : (f[e] = r, f[e].argCount = t);
    }, Dt = (e, r, t) => {
      var n = f["dynCall_" + e];
      return t && t.length ? n.apply(null, [r].concat(t)) : n.call(null, r);
    }, Ae = [], _r, m = (e) => {
      var r = Ae[e];
      return r || (e >= Ae.length && (Ae.length = e + 1), Ae[e] = r = _r.get(e)), r;
    }, kt = (e, r, t) => {
      if (e.includes("j"))
        return Dt(e, r, t);
      var n = m(r).apply(null, t);
      return n;
    }, Ot = (e, r) => {
      var t = [];
      return function() {
        return t.length = 0, Object.assign(t, arguments), kt(e, r, t);
      };
    }, j = (e, r) => {
      e = D(e);
      function t() {
        return e.includes("j") ? Ot(e, r) : m(r);
      }
      var n = t();
      return typeof n != "function" && b(`unknown function pointer with signature ${e}: ${r}`), n;
    }, jt = (e, r) => {
      var t = cr(r, function(n) {
        this.name = r, this.message = n;
        var i = new Error(n).stack;
        i !== void 0 && (this.stack = this.toString() + `
` + i.replace(/^Error(:[^\n]*)?\n/, ""));
      });
      return t.prototype = Object.create(e.prototype), t.prototype.constructor = t, t.prototype.toString = function() {
        return this.message === void 0 ? this.name : `${this.name}: ${this.message}`;
      }, t;
    }, gr, yr = (e) => {
      var r = Dr(e), t = D(r);
      return Y(r), t;
    }, Ee = (e, r) => {
      var t = [], n = {};
      function i(a) {
        if (!n[a] && !Z[a]) {
          if (be[a]) {
            be[a].forEach(i);
            return;
          }
          t.push(a), n[a] = !0;
        }
      }
      throw r.forEach(i), new gr(`${e}: ` + t.map(yr).join([", "]));
    }, Wt = (e, r, t, n, i, a, s, o, u, l, v, h, p) => {
      v = D(v), a = j(i, a), o && (o = j(s, o)), l && (l = j(u, l)), p = j(h, p);
      var $ = lr(v);
      dr($, function() {
        Ee(`Cannot construct ${v} due to unbound types`, [n]);
      }), G([e, r, t], n ? [n] : [], function(w) {
        w = w[0];
        var F, C;
        n ? (F = w.registeredClass, C = F.instancePrototype) : C = Pe.prototype;
        var S = cr($, function() {
          if (Object.getPrototypeOf(this) !== c)
            throw new ee("Use 'new' to construct " + v);
          if (d.constructor_body === void 0)
            throw new ee(v + " has no accessible constructor");
          var Re = d.constructor_body[arguments.length];
          if (Re === void 0)
            throw new ee(`Tried to invoke ctor of ${v} with invalid number of parameters (${arguments.length}) - expected (${Object.keys(d.constructor_body).toString()}) parameters instead!`);
          return Re.apply(this, arguments);
        }), c = Object.create(C, { constructor: { value: S } });
        S.prototype = c;
        var d = new At(v, S, c, p, F, a, o, l);
        d.baseClass && (d.baseClass.__derivedClasses === void 0 && (d.baseClass.__derivedClasses = []), d.baseClass.__derivedClasses.push(d));
        var A = new Te(v, d, !0, !1, !1), E = new Te(v + "*", d, !1, !1, !1), q = new Te(v + " const*", d, !1, !0, !1);
        return fr[e] = { pointerType: E, constPointerType: q }, pr($, S), [A, E, q];
      });
    }, xe = (e, r) => {
      for (var t = [], n = 0; n < e; n++)
        t.push(P[r + n * 4 >> 2]);
      return t;
    };
    function Ye(e, r, t, n, i, a) {
      var s = r.length;
      s < 2 && b("argTypes array size mismatch! Must at least get return value and 'this' types!");
      for (var o = r[1] !== null && t !== null, u = !1, l = 1; l < r.length; ++l)
        if (r[l] !== null && r[l].destructorFunction === void 0) {
          u = !0;
          break;
        }
      var v = r[0].name !== "void", h = s - 2, p = new Array(h), $ = [], w = [];
      return function() {
        arguments.length !== h && b(`function ${e} called with ${arguments.length} arguments, expected ${h}`), w.length = 0;
        var F;
        $.length = o ? 2 : 1, $[0] = i, o && (F = r[1].toWireType(w, this), $[1] = F);
        for (var C = 0; C < h; ++C)
          p[C] = r[C + 2].toWireType(w, arguments[C]), $.push(p[C]);
        var S = n.apply(null, $);
        function c(d) {
          if (u)
            nr(w);
          else
            for (var A = o ? 1 : 2; A < r.length; A++) {
              var E = A === 1 ? F : p[A - 2];
              r[A].destructorFunction !== null && r[A].destructorFunction(E);
            }
          if (v)
            return r[0].fromWireType(d);
        }
        return c(S);
      };
    }
    var Mt = (e, r, t, n, i, a) => {
      var s = xe(r, t);
      i = j(n, i), G([], [e], function(o) {
        o = o[0];
        var u = `constructor ${o.name}`;
        if (o.registeredClass.constructor_body === void 0 && (o.registeredClass.constructor_body = []), o.registeredClass.constructor_body[r - 1] !== void 0)
          throw new ee(`Cannot register multiple constructors with identical number of parameters (${r - 1}) for class '${o.name}'! Overload resolution is currently only performed using the parameter count, not actual type info!`);
        return o.registeredClass.constructor_body[r - 1] = () => {
          Ee(`Cannot construct ${o.name} due to unbound types`, s);
        }, G([], s, (l) => (l.splice(1, 0, null), o.registeredClass.constructor_body[r - 1] = Ye(u, l, null, i, a), [])), [];
      });
    }, mr = (e) => {
      e = e.trim();
      const r = e.indexOf("(");
      return r !== -1 ? (xr(e[e.length - 1] == ")", "Parentheses for argument names should match."), e.substr(0, r)) : e;
    }, Ht = (e, r, t, n, i, a, s, o, u) => {
      var l = xe(t, n);
      r = D(r), r = mr(r), a = j(i, a), G([], [e], function(v) {
        v = v[0];
        var h = `${v.name}.${r}`;
        r.startsWith("@@") && (r = Symbol[r.substring(2)]), o && v.registeredClass.pureVirtualFunctions.push(r);
        function p() {
          Ee(`Cannot call ${h} due to unbound types`, l);
        }
        var $ = v.registeredClass.instancePrototype, w = $[r];
        return w === void 0 || w.overloadTable === void 0 && w.className !== v.name && w.argCount === t - 2 ? (p.argCount = t - 2, p.className = v.name, $[r] = p) : (vr($, r, h), $[r].overloadTable[t - 2] = p), G([], l, function(F) {
          var C = Ye(h, F, v, a, s);
          return $[r].overloadTable === void 0 ? (C.argCount = t - 2, $[r] = C) : $[r].overloadTable[t - 2] = C, [];
        }), [];
      });
    };
    function It() {
      Object.assign($r.prototype, { get(e) {
        return this.allocated[e];
      }, has(e) {
        return this.allocated[e] !== void 0;
      }, allocate(e) {
        var r = this.freelist.pop() || this.allocated.length;
        return this.allocated[r] = e, r;
      }, free(e) {
        this.allocated[e] = void 0, this.freelist.push(e);
      } });
    }
    function $r() {
      this.allocated = [void 0], this.freelist = [];
    }
    var O = new $r(), br = (e) => {
      e >= O.reserved && --O.get(e).refcount === 0 && O.free(e);
    }, Ut = () => {
      for (var e = 0, r = O.reserved; r < O.allocated.length; ++r)
        O.allocated[r] !== void 0 && ++e;
      return e;
    }, xt = () => {
      O.allocated.push({ value: void 0 }, { value: null }, { value: !0 }, { value: !1 }), O.reserved = O.allocated.length, f.count_emval_handles = Ut;
    }, L = { toValue: (e) => (e || b("Cannot use deleted val. handle = " + e), O.get(e).value), toHandle: (e) => {
      switch (e) {
        case void 0:
          return 1;
        case null:
          return 2;
        case !0:
          return 3;
        case !1:
          return 4;
        default:
          return O.allocate({ refcount: 1, value: e });
      }
    } }, Yt = (e, r) => {
      r = D(r), U(e, { name: r, fromWireType: (t) => {
        var n = L.toValue(t);
        return br(t), n;
      }, toWireType: (t, n) => L.toHandle(n), argPackAdvance: x, readValueFromPointer: We, destructorFunction: null });
    }, Ve = (e) => {
      if (e === null)
        return "null";
      var r = typeof e;
      return r === "object" || r === "array" || r === "function" ? e.toString() : "" + e;
    }, Vt = (e, r) => {
      switch (r) {
        case 4:
          return function(t) {
            return this.fromWireType(Ze[t >> 2]);
          };
        case 8:
          return function(t) {
            return this.fromWireType(Ge[t >> 3]);
          };
        default:
          throw new TypeError(`invalid float width (${r}): ${e}`);
      }
    }, Bt = (e, r, t) => {
      r = D(r), U(e, { name: r, fromWireType: (n) => n, toWireType: (n, i) => i, argPackAdvance: x, readValueFromPointer: Vt(r, t), destructorFunction: null });
    }, Lt = (e, r, t, n, i, a, s) => {
      var o = xe(r, t);
      e = D(e), e = mr(e), i = j(n, i), dr(e, function() {
        Ee(`Cannot call ${e} due to unbound types`, o);
      }, r - 1), G([], o, function(u) {
        var l = [u[0], null].concat(u.slice(1));
        return pr(e, Ye(e, l, null, i, a), r - 1), [];
      });
    }, Nt = (e, r, t) => {
      switch (r) {
        case 1:
          return t ? (n) => H[n >> 0] : (n) => k[n >> 0];
        case 2:
          return t ? (n) => ne[n >> 1] : (n) => _e[n >> 1];
        case 4:
          return t ? (n) => R[n >> 2] : (n) => P[n >> 2];
        default:
          throw new TypeError(`invalid integer width (${r}): ${e}`);
      }
    }, zt = (e, r, t, n, i) => {
      r = D(r);
      var a = (v) => v;
      if (n === 0) {
        var s = 32 - 8 * t;
        a = (v) => v << s >>> s;
      }
      var o = r.includes("unsigned"), u = (v, h) => {
      }, l;
      o ? l = function(v, h) {
        return u(h, this.name), h >>> 0;
      } : l = function(v, h) {
        return u(h, this.name), h;
      }, U(e, { name: r, fromWireType: a, toWireType: l, argPackAdvance: x, readValueFromPointer: Nt(r, t, n !== 0), destructorFunction: null });
    }, Xt = (e, r, t) => {
      var n = [Int8Array, Uint8Array, Int16Array, Uint16Array, Int32Array, Uint32Array, Float32Array, Float64Array], i = n[r];
      function a(s) {
        var o = P[s >> 2], u = P[s + 4 >> 2];
        return new i(H.buffer, u, o);
      }
      t = D(t), U(e, { name: t, fromWireType: a, argPackAdvance: x, readValueFromPointer: a }, { ignoreDuplicateRegistrations: !0 });
    }, wr = (e, r, t, n) => {
      if (!(n > 0))
        return 0;
      for (var i = t, a = t + n - 1, s = 0; s < e.length; ++s) {
        var o = e.charCodeAt(s);
        if (o >= 55296 && o <= 57343) {
          var u = e.charCodeAt(++s);
          o = 65536 + ((o & 1023) << 10) | u & 1023;
        }
        if (o <= 127) {
          if (t >= a)
            break;
          r[t++] = o;
        } else if (o <= 2047) {
          if (t + 1 >= a)
            break;
          r[t++] = 192 | o >> 6, r[t++] = 128 | o & 63;
        } else if (o <= 65535) {
          if (t + 2 >= a)
            break;
          r[t++] = 224 | o >> 12, r[t++] = 128 | o >> 6 & 63, r[t++] = 128 | o & 63;
        } else {
          if (t + 3 >= a)
            break;
          r[t++] = 240 | o >> 18, r[t++] = 128 | o >> 12 & 63, r[t++] = 128 | o >> 6 & 63, r[t++] = 128 | o & 63;
        }
      }
      return r[t] = 0, t - i;
    }, Zt = (e, r, t) => wr(e, k, r, t), Cr = (e) => {
      for (var r = 0, t = 0; t < e.length; ++t) {
        var n = e.charCodeAt(t);
        n <= 127 ? r++ : n <= 2047 ? r += 2 : n >= 55296 && n <= 57343 ? (r += 4, ++t) : r += 3;
      }
      return r;
    }, Pr = typeof TextDecoder < "u" ? new TextDecoder("utf8") : void 0, Gt = (e, r, t) => {
      for (var n = r + t, i = r; e[i] && !(i >= n); )
        ++i;
      if (i - r > 16 && e.buffer && Pr)
        return Pr.decode(e.subarray(r, i));
      for (var a = ""; r < i; ) {
        var s = e[r++];
        if (!(s & 128)) {
          a += String.fromCharCode(s);
          continue;
        }
        var o = e[r++] & 63;
        if ((s & 224) == 192) {
          a += String.fromCharCode((s & 31) << 6 | o);
          continue;
        }
        var u = e[r++] & 63;
        if ((s & 240) == 224 ? s = (s & 15) << 12 | o << 6 | u : s = (s & 7) << 18 | o << 12 | u << 6 | e[r++] & 63, s < 65536)
          a += String.fromCharCode(s);
        else {
          var l = s - 65536;
          a += String.fromCharCode(55296 | l >> 10, 56320 | l & 1023);
        }
      }
      return a;
    }, Be = (e, r) => e ? Gt(k, e, r) : "", qt = (e, r) => {
      r = D(r);
      var t = r === "std::string";
      U(e, { name: r, fromWireType(n) {
        var i = P[n >> 2], a = n + 4, s;
        if (t)
          for (var o = a, u = 0; u <= i; ++u) {
            var l = a + u;
            if (u == i || k[l] == 0) {
              var v = l - o, h = Be(o, v);
              s === void 0 ? s = h : (s += String.fromCharCode(0), s += h), o = l + 1;
            }
          }
        else {
          for (var p = new Array(i), u = 0; u < i; ++u)
            p[u] = String.fromCharCode(k[a + u]);
          s = p.join("");
        }
        return Y(n), s;
      }, toWireType(n, i) {
        i instanceof ArrayBuffer && (i = new Uint8Array(i));
        var a, s = typeof i == "string";
        s || i instanceof Uint8Array || i instanceof Uint8ClampedArray || i instanceof Int8Array || b("Cannot pass non-string to std::string"), t && s ? a = Cr(i) : a = i.length;
        var o = Ne(4 + a + 1), u = o + 4;
        if (P[o >> 2] = a, t && s)
          Zt(i, u, a + 1);
        else if (s)
          for (var l = 0; l < a; ++l) {
            var v = i.charCodeAt(l);
            v > 255 && (Y(u), b("String has UTF-16 code units that do not fit in 8 bits")), k[u + l] = v;
          }
        else
          for (var l = 0; l < a; ++l)
            k[u + l] = i[l];
        return n !== null && n.push(Y, o), o;
      }, argPackAdvance: x, readValueFromPointer: hr, destructorFunction(n) {
        Y(n);
      } });
    }, Tr = typeof TextDecoder < "u" ? new TextDecoder("utf-16le") : void 0, Jt = (e, r) => {
      for (var t = e, n = t >> 1, i = n + r / 2; !(n >= i) && _e[n]; )
        ++n;
      if (t = n << 1, t - e > 32 && Tr)
        return Tr.decode(k.subarray(e, t));
      for (var a = "", s = 0; !(s >= r / 2); ++s) {
        var o = ne[e + s * 2 >> 1];
        if (o == 0)
          break;
        a += String.fromCharCode(o);
      }
      return a;
    }, Kt = (e, r, t) => {
      if (t === void 0 && (t = 2147483647), t < 2)
        return 0;
      t -= 2;
      for (var n = r, i = t < e.length * 2 ? t / 2 : e.length, a = 0; a < i; ++a) {
        var s = e.charCodeAt(a);
        ne[r >> 1] = s, r += 2;
      }
      return ne[r >> 1] = 0, r - n;
    }, Qt = (e) => e.length * 2, en = (e, r) => {
      for (var t = 0, n = ""; !(t >= r / 4); ) {
        var i = R[e + t * 4 >> 2];
        if (i == 0)
          break;
        if (++t, i >= 65536) {
          var a = i - 65536;
          n += String.fromCharCode(55296 | a >> 10, 56320 | a & 1023);
        } else
          n += String.fromCharCode(i);
      }
      return n;
    }, rn = (e, r, t) => {
      if (t === void 0 && (t = 2147483647), t < 4)
        return 0;
      for (var n = r, i = n + t - 4, a = 0; a < e.length; ++a) {
        var s = e.charCodeAt(a);
        if (s >= 55296 && s <= 57343) {
          var o = e.charCodeAt(++a);
          s = 65536 + ((s & 1023) << 10) | o & 1023;
        }
        if (R[r >> 2] = s, r += 4, r + 4 > i)
          break;
      }
      return R[r >> 2] = 0, r - n;
    }, tn = (e) => {
      for (var r = 0, t = 0; t < e.length; ++t) {
        var n = e.charCodeAt(t);
        n >= 55296 && n <= 57343 && ++t, r += 4;
      }
      return r;
    }, nn = (e, r, t) => {
      t = D(t);
      var n, i, a, s, o;
      r === 2 ? (n = Jt, i = Kt, s = Qt, a = () => _e, o = 1) : r === 4 && (n = en, i = rn, s = tn, a = () => P, o = 2), U(e, { name: t, fromWireType: (u) => {
        for (var l = P[u >> 2], v = a(), h, p = u + 4, $ = 0; $ <= l; ++$) {
          var w = u + 4 + $ * r;
          if ($ == l || v[w >> o] == 0) {
            var F = w - p, C = n(p, F);
            h === void 0 ? h = C : (h += String.fromCharCode(0), h += C), p = w + r;
          }
        }
        return Y(u), h;
      }, toWireType: (u, l) => {
        typeof l != "string" && b(`Cannot pass non-string to C++ string type ${t}`);
        var v = s(l), h = Ne(4 + v + r);
        return P[h >> 2] = v >> o, i(l, h + 4, v + r), u !== null && u.push(Y, h), h;
      }, argPackAdvance: x, readValueFromPointer: We, destructorFunction(u) {
        Y(u);
      } });
    }, an = (e, r, t, n, i, a) => {
      $e[e] = { name: D(r), rawConstructor: j(t, n), rawDestructor: j(i, a), fields: [] };
    }, sn = (e, r, t, n, i, a, s, o, u, l) => {
      $e[e].fields.push({ fieldName: D(r), getterReturnType: t, getter: j(n, i), getterContext: a, setterArgumentType: s, setter: j(o, u), setterContext: l });
    }, on = (e, r) => {
      r = D(r), U(e, { isVoid: !0, name: r, argPackAdvance: 0, fromWireType: () => {
      }, toWireType: (t, n) => {
      } });
    }, un = {}, fn = (e) => {
      var r = un[e];
      return r === void 0 ? D(e) : r;
    }, Ar = () => {
      if (typeof globalThis == "object")
        return globalThis;
      function e(r) {
        r.$$$embind_global$$$ = r;
        var t = typeof $$$embind_global$$$ == "object" && r.$$$embind_global$$$ == r;
        return t || delete r.$$$embind_global$$$, t;
      }
      if (typeof $$$embind_global$$$ == "object" || (typeof global == "object" && e(global) ? $$$embind_global$$$ = global : typeof self == "object" && e(self) && ($$$embind_global$$$ = self), typeof $$$embind_global$$$ == "object"))
        return $$$embind_global$$$;
      throw Error("unable to get global object.");
    }, ln = (e) => e === 0 ? L.toHandle(Ar()) : (e = fn(e), L.toHandle(Ar()[e])), cn = (e) => {
      e > 4 && (O.get(e).refcount += 1);
    }, Er = (e, r) => {
      var t = Z[e];
      return t === void 0 && b(r + " has unknown type " + yr(e)), t;
    }, vn = (e) => {
      var r = new Array(e + 1);
      return function(t, n, i) {
        r[0] = t;
        for (var a = 0; a < e; ++a) {
          var s = Er(P[n + a * 4 >> 2], "parameter " + a);
          r[a + 1] = s.readValueFromPointer(i), i += s.argPackAdvance;
        }
        var o = new (t.bind.apply(t, r))();
        return L.toHandle(o);
      };
    }, Fr = {}, dn = (e, r, t, n) => {
      e = L.toValue(e);
      var i = Fr[r];
      return i || (i = vn(r), Fr[r] = i), i(e, t, n);
    }, hn = (e, r) => {
      e = Er(e, "_emval_take_value");
      var t = e.readValueFromPointer(r);
      return L.toHandle(t);
    }, pn = () => {
      ae("");
    }, _n = (e, r, t) => k.copyWithin(e, r, r + t), gn = () => 2147483648, yn = (e) => {
      var r = pe.buffer, t = (e - r.byteLength + 65535) / 65536;
      try {
        return pe.grow(t), qe(), 1;
      } catch {
      }
    }, mn = (e) => {
      var r = k.length;
      e >>>= 0;
      var t = gn();
      if (e > t)
        return !1;
      for (var n = (u, l) => u + (l - u % l) % l, i = 1; i <= 4; i *= 2) {
        var a = r * (1 + 0.2 / i);
        a = Math.min(a, e + 100663296);
        var s = Math.min(t, n(Math.max(e, a), 65536)), o = yn(s);
        if (o)
          return !0;
      }
      return !1;
    }, Le = {}, $n = () => de || "./this.program", le = () => {
      if (!le.strings) {
        var e = (typeof navigator == "object" && navigator.languages && navigator.languages[0] || "C").replace("-", "_") + ".UTF-8", r = { USER: "web_user", LOGNAME: "web_user", PATH: "/", PWD: "/", HOME: "/home/web_user", LANG: e, _: $n() };
        for (var t in Le)
          Le[t] === void 0 ? delete r[t] : r[t] = Le[t];
        var n = [];
        for (var t in r)
          n.push(`${t}=${r[t]}`);
        le.strings = n;
      }
      return le.strings;
    }, bn = (e, r) => {
      for (var t = 0; t < e.length; ++t)
        H[r++ >> 0] = e.charCodeAt(t);
      H[r >> 0] = 0;
    }, wn = (e, r) => {
      var t = 0;
      return le().forEach((n, i) => {
        var a = r + t;
        P[e + i * 4 >> 2] = a, bn(n, a), t += n.length + 1;
      }), 0;
    }, Cn = (e, r) => {
      var t = le();
      P[e >> 2] = t.length;
      var n = 0;
      return t.forEach((i) => n += i.length + 1), P[r >> 2] = n, 0;
    }, Pn = (e) => e, Fe = (e) => e % 4 === 0 && (e % 100 !== 0 || e % 400 === 0), Tn = (e, r) => {
      for (var t = 0, n = 0; n <= r; t += e[n++])
        ;
      return t;
    }, Sr = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31], Rr = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31], An = (e, r) => {
      for (var t = new Date(e.getTime()); r > 0; ) {
        var n = Fe(t.getFullYear()), i = t.getMonth(), a = (n ? Sr : Rr)[i];
        if (r > a - t.getDate())
          r -= a - t.getDate() + 1, t.setDate(1), i < 11 ? t.setMonth(i + 1) : (t.setMonth(0), t.setFullYear(t.getFullYear() + 1));
        else
          return t.setDate(t.getDate() + r), t;
      }
      return t;
    };
    function En(e, r, t) {
      var n = t > 0 ? t : Cr(e) + 1, i = new Array(n), a = wr(e, i, 0, i.length);
      return r && (i.length = a), i;
    }
    var Fn = (e, r) => {
      H.set(e, r);
    }, Sn = (e, r, t, n) => {
      var i = P[n + 40 >> 2], a = { tm_sec: R[n >> 2], tm_min: R[n + 4 >> 2], tm_hour: R[n + 8 >> 2], tm_mday: R[n + 12 >> 2], tm_mon: R[n + 16 >> 2], tm_year: R[n + 20 >> 2], tm_wday: R[n + 24 >> 2], tm_yday: R[n + 28 >> 2], tm_isdst: R[n + 32 >> 2], tm_gmtoff: R[n + 36 >> 2], tm_zone: i ? Be(i) : "" }, s = Be(t), o = { "%c": "%a %b %d %H:%M:%S %Y", "%D": "%m/%d/%y", "%F": "%Y-%m-%d", "%h": "%b", "%r": "%I:%M:%S %p", "%R": "%H:%M", "%T": "%H:%M:%S", "%x": "%m/%d/%y", "%X": "%H:%M:%S", "%Ec": "%c", "%EC": "%C", "%Ex": "%m/%d/%y", "%EX": "%H:%M:%S", "%Ey": "%y", "%EY": "%Y", "%Od": "%d", "%Oe": "%e", "%OH": "%H", "%OI": "%I", "%Om": "%m", "%OM": "%M", "%OS": "%S", "%Ou": "%u", "%OU": "%U", "%OV": "%V", "%Ow": "%w", "%OW": "%W", "%Oy": "%y" };
      for (var u in o)
        s = s.replace(new RegExp(u, "g"), o[u]);
      var l = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"], v = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
      function h(c, d, A) {
        for (var E = typeof c == "number" ? c.toString() : c || ""; E.length < d; )
          E = A[0] + E;
        return E;
      }
      function p(c, d) {
        return h(c, d, "0");
      }
      function $(c, d) {
        function A(q) {
          return q < 0 ? -1 : q > 0 ? 1 : 0;
        }
        var E;
        return (E = A(c.getFullYear() - d.getFullYear())) === 0 && (E = A(c.getMonth() - d.getMonth())) === 0 && (E = A(c.getDate() - d.getDate())), E;
      }
      function w(c) {
        switch (c.getDay()) {
          case 0:
            return new Date(c.getFullYear() - 1, 11, 29);
          case 1:
            return c;
          case 2:
            return new Date(c.getFullYear(), 0, 3);
          case 3:
            return new Date(c.getFullYear(), 0, 2);
          case 4:
            return new Date(c.getFullYear(), 0, 1);
          case 5:
            return new Date(c.getFullYear() - 1, 11, 31);
          case 6:
            return new Date(c.getFullYear() - 1, 11, 30);
        }
      }
      function F(c) {
        var d = An(new Date(c.tm_year + 1900, 0, 1), c.tm_yday), A = new Date(d.getFullYear(), 0, 4), E = new Date(d.getFullYear() + 1, 0, 4), q = w(A), Re = w(E);
        return $(q, d) <= 0 ? $(Re, d) <= 0 ? d.getFullYear() + 1 : d.getFullYear() : d.getFullYear() - 1;
      }
      var C = { "%a": (c) => l[c.tm_wday].substring(0, 3), "%A": (c) => l[c.tm_wday], "%b": (c) => v[c.tm_mon].substring(0, 3), "%B": (c) => v[c.tm_mon], "%C": (c) => {
        var d = c.tm_year + 1900;
        return p(d / 100 | 0, 2);
      }, "%d": (c) => p(c.tm_mday, 2), "%e": (c) => h(c.tm_mday, 2, " "), "%g": (c) => F(c).toString().substring(2), "%G": (c) => F(c), "%H": (c) => p(c.tm_hour, 2), "%I": (c) => {
        var d = c.tm_hour;
        return d == 0 ? d = 12 : d > 12 && (d -= 12), p(d, 2);
      }, "%j": (c) => p(c.tm_mday + Tn(Fe(c.tm_year + 1900) ? Sr : Rr, c.tm_mon - 1), 3), "%m": (c) => p(c.tm_mon + 1, 2), "%M": (c) => p(c.tm_min, 2), "%n": () => `
`, "%p": (c) => c.tm_hour >= 0 && c.tm_hour < 12 ? "AM" : "PM", "%S": (c) => p(c.tm_sec, 2), "%t": () => "	", "%u": (c) => c.tm_wday || 7, "%U": (c) => {
        var d = c.tm_yday + 7 - c.tm_wday;
        return p(Math.floor(d / 7), 2);
      }, "%V": (c) => {
        var d = Math.floor((c.tm_yday + 7 - (c.tm_wday + 6) % 7) / 7);
        if ((c.tm_wday + 371 - c.tm_yday - 2) % 7 <= 2 && d++, d) {
          if (d == 53) {
            var E = (c.tm_wday + 371 - c.tm_yday) % 7;
            E != 4 && (E != 3 || !Fe(c.tm_year)) && (d = 1);
          }
        } else {
          d = 52;
          var A = (c.tm_wday + 7 - c.tm_yday - 1) % 7;
          (A == 4 || A == 5 && Fe(c.tm_year % 400 - 1)) && d++;
        }
        return p(d, 2);
      }, "%w": (c) => c.tm_wday, "%W": (c) => {
        var d = c.tm_yday + 7 - (c.tm_wday + 6) % 7;
        return p(Math.floor(d / 7), 2);
      }, "%y": (c) => (c.tm_year + 1900).toString().substring(2), "%Y": (c) => c.tm_year + 1900, "%z": (c) => {
        var d = c.tm_gmtoff, A = d >= 0;
        return d = Math.abs(d) / 60, d = d / 60 * 100 + d % 60, (A ? "+" : "-") + ("0000" + d).slice(-4);
      }, "%Z": (c) => c.tm_zone, "%%": () => "%" };
      s = s.replace(/%%/g, "\0\0");
      for (var u in C)
        s.includes(u) && (s = s.replace(new RegExp(u, "g"), C[u](a)));
      s = s.replace(/\0\0/g, "%");
      var S = En(s, !1);
      return S.length > r ? 0 : (Fn(S, e), S.length - 1);
    }, Rn = (e, r, t, n, i) => Sn(e, r, t, n);
    ir = f.InternalError = class extends Error {
      constructor(r) {
        super(r), this.name = "InternalError";
      }
    }, ct(), ee = f.BindingError = class extends Error {
      constructor(r) {
        super(r), this.name = "BindingError";
      }
    }, Ct(), mt(), Rt(), gr = f.UnboundTypeError = jt(Error, "UnboundTypeError"), It(), xt();
    var Dn = { r: Qr, v: et, a: tt, h: nt, m: it, K: at, R: st, p: ot, ba: ut, d: rt, pa: ft, Y: lt, ga: dt, oa: Wt, na: Mt, E: Ht, fa: Yt, V: Bt, F: Lt, y: zt, t: Xt, U: qt, N: nn, L: an, qa: sn, ha: on, ca: br, ma: ln, W: cn, ka: dn, la: hn, M: pn, ea: _n, da: mn, $: wn, aa: Cn, J: qn, T: si, D: Qn, q: Xn, b: kn, B: Zn, ia: ri, c: Hn, i: Un, j: Wn, o: Kn, Q: Gn, w: zn, I: ni, P: ii, C: ei, H: oi, Z: fi, X: li, k: In, e: Mn, f: jn, g: On, O: ai, l: Ln, ja: Jn, n: xn, S: Yn, u: Bn, x: Nn, A: ti, s: Vn, G: ui, z: Pn, _: Rn }, T = Kr(), Y = f._free = (e) => (Y = f._free = T.ta)(e), Ne = f._malloc = (e) => (Ne = f._malloc = T.ua)(e), Dr = (e) => (Dr = T.wa)(e);
    f.__embind_initialize_bindings = () => (f.__embind_initialize_bindings = T.xa)();
    var _ = (e, r) => (_ = T.ya)(e, r), ce = (e) => (ce = T.za)(e), g = () => (g = T.Aa)(), y = (e) => (y = T.Ba)(e), kr = (e) => (kr = T.Ca)(e), Or = (e) => (Or = T.Da)(e), jr = (e, r, t) => (jr = T.Ea)(e, r, t), Wr = (e) => (Wr = T.Fa)(e);
    f.dynCall_viijii = (e, r, t, n, i, a, s) => (f.dynCall_viijii = T.Ga)(e, r, t, n, i, a, s);
    var Mr = f.dynCall_jiii = (e, r, t, n) => (Mr = f.dynCall_jiii = T.Ha)(e, r, t, n), Hr = f.dynCall_jiiii = (e, r, t, n, i) => (Hr = f.dynCall_jiiii = T.Ia)(e, r, t, n, i);
    f.dynCall_iiiiij = (e, r, t, n, i, a, s) => (f.dynCall_iiiiij = T.Ja)(e, r, t, n, i, a, s), f.dynCall_iiiiijj = (e, r, t, n, i, a, s, o, u) => (f.dynCall_iiiiijj = T.Ka)(e, r, t, n, i, a, s, o, u), f.dynCall_iiiiiijj = (e, r, t, n, i, a, s, o, u, l) => (f.dynCall_iiiiiijj = T.La)(e, r, t, n, i, a, s, o, u, l);
    function kn(e, r) {
      var t = g();
      try {
        return m(e)(r);
      } catch (n) {
        if (y(t), n !== n + 0)
          throw n;
        _(1, 0);
      }
    }
    function On(e, r, t, n) {
      var i = g();
      try {
        m(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function jn(e, r, t) {
      var n = g();
      try {
        m(e)(r, t);
      } catch (i) {
        if (y(n), i !== i + 0)
          throw i;
        _(1, 0);
      }
    }
    function Wn(e, r, t, n, i) {
      var a = g();
      try {
        return m(e)(r, t, n, i);
      } catch (s) {
        if (y(a), s !== s + 0)
          throw s;
        _(1, 0);
      }
    }
    function Mn(e, r) {
      var t = g();
      try {
        m(e)(r);
      } catch (n) {
        if (y(t), n !== n + 0)
          throw n;
        _(1, 0);
      }
    }
    function Hn(e, r, t) {
      var n = g();
      try {
        return m(e)(r, t);
      } catch (i) {
        if (y(n), i !== i + 0)
          throw i;
        _(1, 0);
      }
    }
    function In(e) {
      var r = g();
      try {
        m(e)();
      } catch (t) {
        if (y(r), t !== t + 0)
          throw t;
        _(1, 0);
      }
    }
    function Un(e, r, t, n) {
      var i = g();
      try {
        return m(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function xn(e, r, t, n, i, a) {
      var s = g();
      try {
        m(e)(r, t, n, i, a);
      } catch (o) {
        if (y(s), o !== o + 0)
          throw o;
        _(1, 0);
      }
    }
    function Yn(e, r, t, n, i, a, s) {
      var o = g();
      try {
        m(e)(r, t, n, i, a, s);
      } catch (u) {
        if (y(o), u !== u + 0)
          throw u;
        _(1, 0);
      }
    }
    function Vn(e, r, t, n, i, a, s, o, u, l, v) {
      var h = g();
      try {
        m(e)(r, t, n, i, a, s, o, u, l, v);
      } catch (p) {
        if (y(h), p !== p + 0)
          throw p;
        _(1, 0);
      }
    }
    function Bn(e, r, t, n, i, a, s, o) {
      var u = g();
      try {
        m(e)(r, t, n, i, a, s, o);
      } catch (l) {
        if (y(u), l !== l + 0)
          throw l;
        _(1, 0);
      }
    }
    function Ln(e, r, t, n, i) {
      var a = g();
      try {
        m(e)(r, t, n, i);
      } catch (s) {
        if (y(a), s !== s + 0)
          throw s;
        _(1, 0);
      }
    }
    function Nn(e, r, t, n, i, a, s, o, u) {
      var l = g();
      try {
        m(e)(r, t, n, i, a, s, o, u);
      } catch (v) {
        if (y(l), v !== v + 0)
          throw v;
        _(1, 0);
      }
    }
    function zn(e, r, t, n, i, a, s) {
      var o = g();
      try {
        return m(e)(r, t, n, i, a, s);
      } catch (u) {
        if (y(o), u !== u + 0)
          throw u;
        _(1, 0);
      }
    }
    function Xn(e) {
      var r = g();
      try {
        return m(e)();
      } catch (t) {
        if (y(r), t !== t + 0)
          throw t;
        _(1, 0);
      }
    }
    function Zn(e, r, t, n) {
      var i = g();
      try {
        return m(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function Gn(e, r, t, n, i, a, s) {
      var o = g();
      try {
        return m(e)(r, t, n, i, a, s);
      } catch (u) {
        if (y(o), u !== u + 0)
          throw u;
        _(1, 0);
      }
    }
    function qn(e, r, t, n) {
      var i = g();
      try {
        return m(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function Jn(e, r, t, n, i, a, s, o) {
      var u = g();
      try {
        m(e)(r, t, n, i, a, s, o);
      } catch (l) {
        if (y(u), l !== l + 0)
          throw l;
        _(1, 0);
      }
    }
    function Kn(e, r, t, n, i, a) {
      var s = g();
      try {
        return m(e)(r, t, n, i, a);
      } catch (o) {
        if (y(s), o !== o + 0)
          throw o;
        _(1, 0);
      }
    }
    function Qn(e, r, t, n, i, a) {
      var s = g();
      try {
        return m(e)(r, t, n, i, a);
      } catch (o) {
        if (y(s), o !== o + 0)
          throw o;
        _(1, 0);
      }
    }
    function ei(e, r, t, n, i, a, s, o, u, l) {
      var v = g();
      try {
        return m(e)(r, t, n, i, a, s, o, u, l);
      } catch (h) {
        if (y(v), h !== h + 0)
          throw h;
        _(1, 0);
      }
    }
    function ri(e, r, t) {
      var n = g();
      try {
        return m(e)(r, t);
      } catch (i) {
        if (y(n), i !== i + 0)
          throw i;
        _(1, 0);
      }
    }
    function ti(e, r, t, n, i, a, s, o, u, l) {
      var v = g();
      try {
        m(e)(r, t, n, i, a, s, o, u, l);
      } catch (h) {
        if (y(v), h !== h + 0)
          throw h;
        _(1, 0);
      }
    }
    function ni(e, r, t, n, i, a, s, o) {
      var u = g();
      try {
        return m(e)(r, t, n, i, a, s, o);
      } catch (l) {
        if (y(u), l !== l + 0)
          throw l;
        _(1, 0);
      }
    }
    function ii(e, r, t, n, i, a, s, o, u) {
      var l = g();
      try {
        return m(e)(r, t, n, i, a, s, o, u);
      } catch (v) {
        if (y(l), v !== v + 0)
          throw v;
        _(1, 0);
      }
    }
    function ai(e, r, t, n, i, a, s) {
      var o = g();
      try {
        m(e)(r, t, n, i, a, s);
      } catch (u) {
        if (y(o), u !== u + 0)
          throw u;
        _(1, 0);
      }
    }
    function si(e, r, t, n) {
      var i = g();
      try {
        return m(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function oi(e, r, t, n, i, a, s, o, u, l, v, h) {
      var p = g();
      try {
        return m(e)(r, t, n, i, a, s, o, u, l, v, h);
      } catch ($) {
        if (y(p), $ !== $ + 0)
          throw $;
        _(1, 0);
      }
    }
    function ui(e, r, t, n, i, a, s, o, u, l, v, h, p, $, w, F) {
      var C = g();
      try {
        m(e)(r, t, n, i, a, s, o, u, l, v, h, p, $, w, F);
      } catch (S) {
        if (y(C), S !== S + 0)
          throw S;
        _(1, 0);
      }
    }
    function fi(e, r, t, n) {
      var i = g();
      try {
        return Mr(e, r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function li(e, r, t, n, i) {
      var a = g();
      try {
        return Hr(e, r, t, n, i);
      } catch (s) {
        if (y(a), s !== s + 0)
          throw s;
        _(1, 0);
      }
    }
    var Se;
    ie = function e() {
      Se || Ir(), Se || (ie = e);
    };
    function Ir() {
      if (z > 0 || (Yr(), z > 0))
        return;
      function e() {
        Se || (Se = !0, f.calledRun = !0, !Xe && (Vr(), B(f), f.onRuntimeInitialized && f.onRuntimeInitialized(), Br()));
      }
      f.setStatus ? (f.setStatus("Running..."), setTimeout(function() {
        setTimeout(function() {
          f.setStatus("");
        }, 1), e();
      }, 1)) : e();
    }
    if (f.preInit)
      for (typeof f.preInit == "function" && (f.preInit = [f.preInit]); f.preInit.length > 0; )
        f.preInit.pop()();
    return Ir(), V.ready;
  };
})();
function yi(W) {
  return ci(De, W);
}
async function mi(W, {
  tryHarder: V = N.tryHarder,
  formats: f = N.formats,
  maxSymbols: B = N.maxSymbols
} = N) {
  return vi(
    W,
    {
      tryHarder: V,
      formats: f,
      maxSymbols: B
    },
    De
  );
}
async function $i(W, {
  tryHarder: V = N.tryHarder,
  formats: f = N.formats,
  maxSymbols: B = N.maxSymbols
} = N) {
  return di(
    W,
    {
      tryHarder: V,
      formats: f,
      maxSymbols: B
    },
    De
  );
}
async function bi(W, {
  format: V = J.format,
  charset: f = J.charset,
  quietZone: B = J.quietZone,
  width: K = J.width,
  height: ve = J.height,
  eccLevel: de = J.eccLevel
} = J) {
  return hi(
    W,
    {
      format: V,
      charset: f,
      quietZone: B,
      width: K,
      height: ve,
      eccLevel: de
    },
    De
  );
}
export {
  Pi as ZXING_BARCODE_FORMAT_NAMES,
  Ti as ZXING_CHARACTOR_SET_NAMES,
  N as defaultZXingReadOptions,
  J as defaultZXingWriteOptions,
  yi as getZXingModule,
  Ai as purgeZXingModule,
  $i as readBarcodeFromImageData,
  mi as readBarcodesFromImageFile,
  Ei as setZXingModuleOverrides,
  bi as writeBarcodeToImageFile
};
